modules/core/src/main/java/org/apache/ignite/internal/processors/cache/checker/processor/PartitionReconciliationProcessor.java

```diff
@@ -0,0 +1,600 @@
+     * doubtful keys.
+     */
+    private final RepairAlgorithm repairAlg;
+
+    /**
+     * Keys that were detected as incosistent during the reconciliation process.
+     */
+    private final Map<String, Map<Integer, List<PartitionReconciliationDataRowMeta>>> inconsistentKeys = new HashMap<>();
+
+    /**
+     * Entries that were detected as inconsistent but weren't repaired due to some reason.
+     */
+    private final Map<String, Map<Integer, Set<PartitionReconciliationSkippedEntityHolder<PartitionReconciliationKeyMeta>>>> skippedEntries = new HashMap<>();
+
+    /**
+     * Progress tracker.
+     */
+    private final WorkProgress workProgress = new WorkProgress();
+
+    /**
+     * Creates a new instance of Partition reconciliation processor.
+     *
+     * @param sesId Session identifier that allows to identify different runs of the utility.
+     * @param ignite Local Ignite instance to be used as an entry point for the execution of the utility.
+     * @param caches Collection of cache names to be checked.
+     * @param fixMode Flag indicates that inconsistencies should be repaired.
+     * @param parallelismLevel Number of batches that can be handled simultaneously.
+     * @param batchSize Amount of keys to retrieve within one job.
+     * @param recheckAttempts Amount of potentially inconsistent keys recheck attempts.
+     * @param repairAlg Repair algorithm to be used to fix inconsistency.
+     * @param recheckDelay Specifies the time interval between two consequent attempts to check keys.
+     */
+    @SuppressWarnings("AssignmentOrReturnOfFieldWithMutableType")
+    public PartitionReconciliationProcessor(
+        long sesId,
+        IgniteEx ignite,
+        Collection<String> caches,
+        boolean fixMode,
+        int parallelismLevel,
+        int batchSize,
+        int recheckAttempts,
+        RepairAlgorithm repairAlg,
+        int recheckDelay
+    ) throws IgniteCheckedException {
+        super(sesId, ignite, parallelismLevel);
+
+        this.recheckDelay = recheckDelay;
+        this.caches = caches;
+        this.fixMode = fixMode;
+        this.batchSize = batchSize;
+        this.recheckAttempts = recheckAttempts;
+        this.repairAlg = repairAlg;
+    }
+
+    /**
+     * @return Partition reconciliation result
+     */
+    public ExecutionResult<PartitionReconciliationResult> execute() {
+        log.info(String.format(START_EXECUTION_MSG, fixMode, repairAlg, batchSize, recheckAttempts, parallelismLevel, caches));
+
+        try {
+            for (String cache : caches) {
+                IgniteInternalCache<Object, Object> cachex = ignite.cachex(cache);
+
+                Factory expiryPlcFactory = cachex.configuration().getExpiryPolicyFactory();
+                if (expiryPlcFactory != null && !(expiryPlcFactory.create() instanceof EternalExpiryPolicy)) {
+                    log.warning("The cache '" + cache + "' was skipped because CacheConfiguration#setExpiryPolicyFactory exists.");
```

Comment:
`CacheConfiguration#setExpiryPolicyFactory` method exists independently of expiry policy :) Please, reformulate message.